library/src/scripts/flyouts/NewPostMenu.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { ReactNode, useState } from "react";
+import classNames from "classnames";
+
+import { NewPostMenuIcon } from "@library/icons/common";
+import LinkAsButton from "@library/routing/LinkAsButton";
+import Button from "@library/forms/Button";
+import { ButtonTypes } from "@library/forms/buttonTypes";
+import { newPostMenuClasses } from "@library/flyouts/newPostMenuStyles";
+import { Trail } from "react-spring/renderprops";
+import { useSpring, animated, interpolate } from "react-spring";
+import NewPostBackground from "./NewPostBackground";
+
+export enum PostTypes {
+    LINK = "link",
+    BUTTON = "button",
+}
+
+export interface IAddPost {
+    id: string;
+    action: (() => void) | string;
+    type: PostTypes;
+    className?: string;
+    label: string;
+    icon: JSX.Element;
```

Comment:
```suggestion
    icon: React.ReactNode;
```

If you're looking to reflect something renderable by react, this is a more accurate type (supports fragments, node arrays, strings, null, numbers, etc).